src/actions.ml

```diff
@@ -61,40 +79,78 @@ let send_status_check ~bot_info job_info pr_num (gh_owner, gh_repo)
   else
     Lwt_io.printf "Pushing a status check...\n"
     >>= fun () ->
-    GitHub_mutations.send_status_check ~repo_full_name:github_repo_full_name
-      ~commit:job_info.commit ~state:"failure"
-      ~url:
-        (Printf.sprintf "https://gitlab.com/%s/-/jobs/%d" repo_full_name
-           job_info.build_id)
-      ~context
-      ~description:(failure_reason ^ " on GitLab CI")
-      ~bot_info
+    match bot_info.github_token with
+    | ACCESS_TOKEN _t ->
+        GitHub_mutations.send_status_check ~repo_full_name:github_repo_full_name
+          ~commit:job_info.commit ~state:"failure" ~url:job_url ~context
+          ~description:(failure_reason ^ " on GitLab CI")
+          ~bot_info
+    | INSTALL_TOKEN _t -> (
+        GitHub_queries.get_repository_id ~bot_info ~owner:gh_owner ~repo:gh_repo
+        >>= function
+        | Ok repo_id ->
+            GitHub_mutations.create_check_run ~bot_info ~name:context ~repo_id
+              ~head_sha:job_info.commit ~conclusion:FAILURE ~status:COMPLETED
+              ~title:(failure_reason ^ " on GitLab CI")
+              ~text:job_url ~details_url:job_url ~summary:job_url
+        | Error _ ->
+            Lwt.return () )
```

Comment:
It seems to me that the exact same logic is repeated often, so it could be convenient to move it to an external function and reuse it where convenient (not necessarily in places where there are small variations / deviations from this exact code / logic).

Regarding the last two lines, why not print something in the log when that happens? (In some other places, you already do it BTW.)